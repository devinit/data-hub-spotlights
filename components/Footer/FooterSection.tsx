import React, { FunctionComponent } from 'react';
import { FooterSection as FooterSectionType, NavigationItem, SocialLink } from '../DefaultLayout';

export type FooterSectionProps = FooterSectionType & { primaryNavigation: NavigationItem[] };
interface SectionLinkProps {
  url: string;
  label?: string;
}

export const FooterSectionLink: FunctionComponent<SectionLinkProps> = ({ children, url, label }) =>
  <li><a href={ url }>{ label || children }</a></li>;

export const FooterSocialLink: FunctionComponent<SocialLink> = ({ social_platform, link_url, image_url }) =>
  <FooterSectionLink url={ link_url }>
    <img src={ image_url } alt={ social_platform }/>
  </FooterSectionLink>;

const FooterSection: FunctionComponent<FooterSectionProps> = props => {
  return (
    <div className="l-footer__col">
      <h4 className="footer__title">{ props.title }</h4>
      <ul className="footer__list">
        {
          props.primaryNavigation.map(({ full_url, title }, index) =>
            <FooterSectionLink key={ `${title} ${index}` } url={ full_url } label={ title }/>
          )
        }
        {
          props.section_links.map(({ link_url, page_url, label }, index) =>
            <FooterSectionLink key={ `${label} ${index}` } url={ link_url || page_url } label={ label }/>
          )
        }
      </ul>
      <ul className="footer__social">
        {
          props.social_links.map(({ link_url, social_platform, image_url }, index) =>
            <FooterSocialLink
              key={ `${social_platform} ${index}` }
              link_url={ link_url }
              image_url={ image_url }
              social_platform={ social_platform }
            />
          )
        }
      </ul>
    </div>
  );
};

export { FooterSection };
