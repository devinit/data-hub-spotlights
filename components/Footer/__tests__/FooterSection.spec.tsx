/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { FooterSection, FooterSectionLink, FooterSocialLink } from '../FooterSection';
import { section, sectionLinks, socialLinks } from './mockData';

describe('FooterSection', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<FooterSection {...section} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders either a page_url or a link_url', () => {
    sectionLinks[0].page_url = 'http://localhost:3000/my-page';
    sectionLinks[0].link_url = '';
    const renderer = TestRenderer.create(<FooterSection {...section} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});

describe('FooterSectionLink', () => {
  test('renders correctly', () => {
    const sectionLink = sectionLinks[0];
    const renderer = TestRenderer.create(
      <FooterSectionLink url={sectionLink.link_url} label={sectionLink.label} />
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});

describe('FooterSocialLink', () => {
  test('renders correctly', () => {
    const socialLink = socialLinks[0];
    const renderer = TestRenderer.create(<FooterSocialLink {...socialLink} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
