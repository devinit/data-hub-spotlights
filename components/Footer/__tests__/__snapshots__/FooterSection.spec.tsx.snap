// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterSection renders correctly 1`] = `
<div
  className="l-footer__col"
>
  <h4
    className="footer__title"
  >
    More links
  </h4>
  <ul
    className="footer__list"
  >
    <li>
      <a
        href="https://localhost:3000/nav-1"
      >
        Nav 1
      </a>
    </li>
    <li>
      <a
        href="https://localhost:3000/nav-2"
      >
        Nav 2
      </a>
    </li>
    <li>
      <a
        href="http://localhost:3000/contact-us"
      >
        Contact us
      </a>
    </li>
  </ul>
  <ul
    className="footer__social"
  >
    <li>
      <a
        href="http://localhost:3000/contact-us"
      >
        <img
          alt="facebook"
          src="/assets/http://localhost:3000/assets/images/image.svg"
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`FooterSection renders either a page_url or a link_url 1`] = `
<div
  className="l-footer__col"
>
  <h4
    className="footer__title"
  >
    More links
  </h4>
  <ul
    className="footer__list"
  >
    <li>
      <a
        href="https://localhost:3000/nav-1"
      >
        Nav 1
      </a>
    </li>
    <li>
      <a
        href="https://localhost:3000/nav-2"
      >
        Nav 2
      </a>
    </li>
    <li>
      <a
        href="http://localhost:3000/my-page"
      >
        Contact us
      </a>
    </li>
  </ul>
  <ul
    className="footer__social"
  >
    <li>
      <a
        href="http://localhost:3000/contact-us"
      >
        <img
          alt="facebook"
          src="/assets/http://localhost:3000/assets/images/image.svg"
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`FooterSectionLink renders correctly 1`] = `
<li>
  <a
    href=""
  >
    Contact us
  </a>
</li>
`;

exports[`FooterSocialLink renders correctly 1`] = `
<li>
  <a
    href="http://localhost:3000/contact-us"
  >
    <img
      alt="facebook"
      src="http://localhost:3000/assets/images/image.svg"
    />
  </a>
</li>
`;
